### PR TITLE
Add support for the Java API

### PR DESCRIPTION
**Pull request by jdubois**
This is copied from https://github.com/Azure/azure-dev-pr/pull/716 as asked by @savannahostrowski

As this was coded 6 months ago, I guess the codebase evolved, so at the moment this is mostly for not losing the existing code, but this needs to be updated.

### CODE REVIEW

**Comment by jdubois** [comment]
@roryp and @bbenz - can you have a look at my PR? I would love to have reviews from both of you

File: templates/todo/projects/java-mongo/azure.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+name: todo-nodejs-mongo
+services:
+  - name: ${BASENAME}api
+    project: ../../../api/java
+    language: java
```
Comment (akaroml) [inline templates/todo/projects/java-mongo/azure.yaml:5]: Were you able to get through with this?

**Comment by weikanglim** [comment]
Hey @jdubois, heads up that I'm the azd developer that will be looking into enabling Java support for `azd` and I'll likely start by updating the template based on recent repository changes.

**Comment by jdubois** [comment]
Hey @weikanglim ! Don't hesitate to ping me, or people in my team (most noticeably @roryp and @bbenz ), we'll be happy to help.

**Comment by weikanglim** [comment]
Closing this PR since this work has been incorporated into #799

**Pull request closed without merging**